src/swupd.h

```diff
@@ -164,6 +164,8 @@ extern int get_absolute_latest_version(void);
 extern enum swupd_code read_versions(int *current_version, int *server_version, char *path_prefix);
 extern int get_current_version(char *path_prefix);
 extern bool get_distribution_string(char *path_prefix, char *dist);
+extern void get_current_format(char **current_format);
```

Comment:
Use the same API you use for get_latest_format() for consistency, returning an int instead of saving it in a char.
If it's a negative number you can print "unknown"

Reply:
Done